app/controllers/tweet_controller.rb

```diff
@@ -0,0 +1,92 @@
+require 'sinatra/base'
+require 'rack-flash'
+class TweetController < ApplicationController
+
+    enable :sessions
+    enable :method_override
+
+    use Rack::Flash
+ 
+    get '/tweets' do
+        if logged_in?(session)
+            @user = current_user(session)
+            @tweets = Tweet.all 
+            @users = User.all
+        else
+            redirect to "/login"
+        end
+        erb :'tweets/tweets'
+    end
+
+    get '/tweets/new' do
+        if logged_in?(session)
+            erb :'tweets/create_tweet'
+        else
+            redirect to "/login"
+        end
+    end
+
+    post '/tweets' do
+        @tweet = Tweet.new(params)
+
+        if @tweet.content == ""
+            redirect to "/tweets/new"
+        elsif @tweet.save
+            @user = current_user(session)
+            @user.tweets << @tweet
+            redirect to "/tweets"
+         else
+            redirect to "/signup"
+        end
+    end 
+
+    get '/tweets/:id' do
+        if logged_in?(session)
+            @tweet = current_tweet(params[:id])
+            erb :'tweets/show_tweet'
+        else
+            redirect to "/login"
+        end
+    end
+
+    get '/tweets/:id/edit' do
+        if logged_in?(session)
+            @tweet = current_tweet(params[:id])
```

Comment:
Shouldn't you verify a tweet belongs to the `current_user` (and not just that any user is logged in) before allowing them to edit?